### PR TITLE
Christopher Beards

### PR DESCRIPTION
**Pull request by ChristopherBeards**
## Trello Set Up:
* [x] Use your existing Trello account from the Front End Project, or create a new one
* [x] Create a new board called "Lambda Notes(Backend) - {Your Name}"
   - https://trello.com/b/1xl0u3rs
* [x] Create lists titled `backlog`,`To Do`, `In Progress`, and `Done`
* [x] Fill in the `To Do` list with the MVP features listed below
* [x] Fill in the `backlog` list with all the extra features listed below
* [x] Share your board with the project manager that has been assigned to you.  If you have not been assigned yet, reach out to your Section Lead for guidance
* [x] Add your Trello URL to your project's README.md file.  Commit the change, push it to your repository & submit a pull request

## Backend MVP Features:
* [x] Create a Database for your app
* [x] Create a Node app and connect it to your database
* [x] Create a Registration Page that allows users to create accounts for your app and sign in with email/password
* [x] Store these User Accounts in your Database
* [x] Add a Sign In Page
* [x] Store Notes in your Database

* [x] Display a list of notes
* [x] Create a note with a title and content
* [x] View an existing note
* [x] Edit an existing note
* [x] Delete an existing note
* [x] Write tests for your endpoints
## Deploy your application to the web:
* [x] Front-End
* [ ] Server
* [ ] DB

# https://cmb-lambdanotes.netlify.com/

* Set local routes: 
Create A User - `/api/users`
Login - `/api/login`
Post New Note - `/api/new-note`
Delete Note - `/api/note/delete/:id`
Get All Notes - `/api/notes/:username`
Get Single Note - `/api/note/:id`
Edit Note - `/api/note/edit/:id`

## Extra Features:
* [x] Allow users to clone notes
* [ ] Allow users to sign in with a third party service (google, facebook, github, club penguin, etc...)
* [ ] Search functionality
* [ ] Allow multiple users to collaborate on notes
* [ ] Add pagination for long lists of notes
* [ ] Create and display tags that can be added to notes and stored in the Database
* [ ] Allow users to attach images to notes
* [ ] Allow users to create Lists and assign notes to a list
* [ ] Setup Auto-Deploy on Heroku

## Super Duper Extra Credit Bonus Features
* [ ] Add a payment form integrating with Stripe that allows Users to buy a "Premium" version of LambdaNotes.
* [ ] Gate your favorite feature behind the premium paywall

### CODE REVIEW

**Comment by wildpow** [comment]
Everything looks good. Let me know when you get to the extra credit.

**Comment by wildpow** [comment]
Git commit messages look great. Be sure to add a link to your deployed app in your readme.

File: lambda-notes/src/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+export const AUTHENTICATION_ERROR = 'AUTHENTICATION_ERROR';
+export const GET_USERS = 'GET_USERS';
+export const CHECK_IF_AUTHENTICATED = 'CHECK_IF_AUTHENTICATED';
+export const GET_NOTES = 'GET_NOTES';
+export const ABC_SORT = 'ABC_SORT';
+
+// axios.defaults.withCredentials = true;
+
+//=====================================
+//         Validation Actions
```
Comment (Velsu) [inline lambda-notes/src/actions/index.js]: Actions file concise and separated by categories, very nice.

File: lambda-notes/src/components/DisplayOne.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    this.props.deleteNote(this.state.note._id, this.props.history);
+  };
+
+  toggleEdit = () => {
+    this.setState({ edited: !this.state.edited });
+  };
+
+  createClone = e => {
+    e.preventDefault();
+    const cloned = {
```
Comment (Velsu) [inline lambda-notes/src/components/DisplayOne.js]: Good job on cloning the notes.

File: lambda-notes/src/components/CreateNote.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                  type="text"
+                  className="titleField"
+                  placeholder="Title"
+                  name="title"
+                  onChange={this.handleFormChange}
+                  value={this.state.title}
+                />
+              </div>
+              <div className="bodyContainer">
+                <textarea
```
Comment (Velsu) [inline lambda-notes/src/components/CreateNote.js]: Can add `maxlength` to textarea to limit number of characters.

File: lambda-notes/src/reducers/index.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      return {
+        ...state,
+        notes: state.notes.map(note => {
+          if (note.id !== action.payload.id) return note;
+          return action.payload;
+        })
+      };
+      case ABC_SORT:
+      return {
+        ...state,
```
Comment (Velsu) [inline lambda-notes/src/reducers/index.js]: Great job on getting sorting to work.

**Pull request closed without merging**